### DISCUSSION TITLE
ERROR Message ：There was a timeout in accessing the Microsoft OneDrive service

### DISCUSSION THREAD

**Discussion by leon-zhang-guoao** [abraunegg/onedrive#1571] [Q&A] [answered]
Hi
This error happens quite often and I would like to know if there is any way to fix it.
Error Message: There was a timeout in accessing the Microsoft OneDrive service - Internet connectivity issue?
onedrive v2.4.12

Thanks

**Comment by abraunegg** [comment, marked as answer]
> Hi
> This error happens quite often and I would like to know if there is any way to fix it.
> Error Message: There was a timeout in accessing the Microsoft OneDrive service - Internet connectivity issue?
> onedrive v2.4.12
> 
> Thanks

This is your clue to how to fix this:

>     `There was a timeout in accessing the Microsoft OneDrive service - Internet connectivity issue?`


This means that:
* You have a poor / unstable Internet connection
* You have poor DNS resolution

Actions for you:
* Review your network / network access to the Internet
* Review https://github.com/abraunegg/onedrive/wiki/Frequently-Asked-Questions#cannot-connect-to-microsoft-onedrive-service---network-connection-issue
* Change ISP's

**Reply by leon-zhang-guoao** [reply]
I run the 'curl -i - connect timeout - 5 - Max - time 5 https://login.microsoftonline.com' this command, return is normal.
My previous OneDrive version was 2.4.0, which was always stable, but after I upgraded it to 2.4.12 last week, this issue has been cropping up quite often, so I don't think it's a problem with my network.

**Reply by abraunegg** [reply]
@leon-zhang-guoao 
> My previous OneDrive version was 2.4.0, which was always stable, but after I upgraded it to 2.4.12 last week, this issue has been cropping up quite often, so I don't think it's a problem with my network.

Nothing in the codebase has changed, in regards to network | timeout configuration - only better reporting / output of when the issue is occurring. This is maybe why you are *only* seeing this now.

The error message is your clue on how to solve. The client, was unable to access OneDrive .. This is not an issue with the client, but something inbetween you and OneDrive.

Are you running Ubuntu? If yes, start by disabling your 'ufw' .. as this has been known to cause issues with networking applications

Another suggestion is - if you are seeing this when running as a systemd service or manually via --monitor, change the 'monitor_interval' value to 200 as this appears to resolve other Ubuntu connectivity issues.

If either of these work - then - this is *proof* that it is not the client that is at fault - but something on your local system causing the issue .........

**Reply by leon-zhang-guoao** [reply]
Thank you very much, let me try as you said.
I wish you a great week.

**Reply by leon-zhang-guoao** [reply]
I tried the method you mentioned. It still didn't work. I think my environment is fine because after a problem, a reboot will restore it to normal. If it's a problem with the environment, a reboot shouldn't solve the problem.

**Reply by abraunegg** [reply]
@leon-zhang-guoao 
> I tried the method you mentioned. It still didn't work. I think my environment is fine because after a problem, a reboot will restore it to normal. If it's a problem with the environment, a reboot shouldn't solve the problem.

So a reboot of your Linux system fixes the issue for a certain period of time .... this just reinforces the issue is with your Linux system ...

Are you using Ubuntu? If yes, install a totally different Linux distribution (dont use Ubuntu) - and not in a Virtual Machine either .. and see if you can replicate the issue in a different Linux distribution .. 

If you cant .. then this is an issue with your chosen Linux distribution - not this application .....

**Reply by leon-zhang-guoao** [reply]
I rebooted OneDrive not Linux system...

**Reply by abraunegg** [reply]
@leon-zhang-guoao 

Even so - when you 'restart' the application you create a new TCP connection to OneDrive. The application uses your Linux OS / Linux TCP stack to communicate - the onedrive client does not have its own TCP stack

So if something is failing after a few minutes after having established successful communication - *something* on your Linux system is *causing* this .. not the onedrive application.

To validate this, if you are using Ubuntu, install a whole new Linux distribution that is *not* Ubuntu based ... does the same issue occur?

I can 99% tell you - this is something entirely environmental to:
* Your Linux OS
* Your network
* Your Internet access to OneDrive

**Reply by leon-zhang-guoao** [reply]
All right. Thank you. I'll try it on.
